src/librustdoc/test.rs

```diff
@@ -410,12 +450,28 @@ impl Collector {
             current_header: None,
             cratename: cratename,
             opts: opts,
+            code_blocks: block_lines,
+            filename: filename,
+        }
+    }
+
+    fn get_line_from_key(&mut self, key: &String) -> u32 {
+        let (line, need_removal) = if let Some(l) = self.code_blocks.get_mut(key) {
+            let need_removal = l.len() > 1;
+            (l.pop().unwrap_or(1), need_removal)
+        } else {
+            return 1;
+        };
+        if need_removal {
+            self.code_blocks.remove(key);
         }
+        line
```

Comment:
The hash map contains all block codes and they key is the "id" of this block code (understand: the order it appears in the file). I did it this way to be able to print a code block more easily. You can see its creation [here](https://github.com/rust-lang/rust/pull/36320/files#diff-beabfb1be9404d986c165f4fdd92285cR431).